index.d.ts

```diff
@@ -96,6 +96,9 @@ declare namespace Cmd {
   export const dispatch: symbol;
   export const getState: symbol;
   export const none: NoneCmd;
+  export type Dispatch = <A extends Action>(a: A) => void;
```

Comment:
dispatch technically returns a promise

Reply:
A promise of the action or void, ? `<A extends Action>(a: A) => Promise<A>`. The redux typing returns just the action:
```
export interface Dispatch<A extends Action = AnyAction> {
  <T extends A>(action: T): T
}
```

Reply:
Why do we define and export `Dispatch` in here at all? Let's just import it from Redux.

Reply:
@domarmstrong @laszlopandy any thoughts on this? is the reason we don't just reuse the redux dispatch type because the return type is different?

and @domarmstrong i think i forgot to respond to your original comment months ago.  it actually returns a promise with the new state object.

Reply:
I think we should have a different type because the Redux type definitions show that dispatch returns an action, but in redux-loop we return a `Promise<Action>`. I think this warrants making a new type, but yes we should be consistent about this. 
Also I never understood why dispatch would return anything in redux. If someone knows, I would be glad to understand it.